### PR TITLE
bump sbt version

### PR DESCRIPTION
**Pull request by joprice**

### CODE REVIEW

**Comment by stanch** [comment]
Thanks! Could you rewrite the commit message according to https://github.com/47deg/macroid#commit-messages? While we are at it, maybe it makes sense to also bump `android-sdk-plugin`...

**Comment by joprice** [comment]
I have another commit I was going to follow up with that bumps multiple library versions, the android plugin, adds scoverage, and switches the main scala version to 2.11 in order to work with scoverage. Do you want me to merge that all into this one?

**Comment by stanch** [comment]
That sounds reasonable, yes.

**Comment by joprice** [comment]
Looks like it failed on the target platform update. I changed that locally since I didn't have 21 installed. Do we want to upgrade the platform used on travis or stick with 21?

File: build.sbt [hunk 1]
```diff
@@ -39,8 +41,8 @@ lazy val core = (project in file("macroid-core"))
     unmanagedSourceDirectories in Test := Seq(baseDirectory.value / "src" / "test" / "scala"),
 
     libraryDependencies ++= Seq(
-      "com.android.support" % "support-v4" % "21.0.3",
-      "org.scala-lang.modules" %% "scala-async" % "0.9.4",
+      "com.android.support" % "support-v4" % "23.1.1",
+      "org.scala-lang.modules" %% "scala-async" % "0.9.5",
       "org.scalatest" %% "scalatest" % "2.1.5" % "test"
```
Comment (stanch) [inline build.sbt]: This can also be updated.

**Comment by stanch** [comment]
Sure, let’s update the Travis configuration :)

**Comment by stanch** [comment]
Apparently now the build tools are just “tools”:
```
id: 1 or "tools"
     Type: Tool
     Desc: Android SDK Tools, revision 24.4.1
```
So it should be `--filter tools`.

**Comment by joprice** [comment]
Just 'tools' fails for me as well. `Error: Ignoring unknown package filter 'tools'`

**Comment by joprice** [comment]
CircleCI lets you ssh onto the box to debug when a build fails. Does travis have a similar feature? I'd like to try running `android list sdk --extended` and see what's available.

**Comment by stanch** [comment]
I don’t think so, but you can always include that line in the file just to test it :)

**Comment by joprice** [comment]
Feels like debugging IE6 :-)

**Comment by stanch** [comment]
Well, `tools` is there :) Do you have the same version of the SDK on your machine?

**Comment by joprice** [comment]
That one passed. I pushed another with a cache config in travis.yml, so hopefully the next is a bit quicker.

**Comment by joprice** [comment]
Pushing one more to see if the cache is hit.

**Comment by joprice** [comment]
All done

**Comment by stanch** [comment]
Awesome! I just discovered that the same error message is apparently printed when the package is already installed o_O. That explains a lot... Do you think it’s worth it to cache the `android-sdk-linux` directory as well? (In this case the `wget` will only need to run if that directory is not present.

**Comment by joprice** [comment]
Yes I noticed that locally. It only searches the list of available updates. That's why I had trouble verifying locally, since I had that exact version installed. We probably could cache that. Something like this:

```bash
[ ! -f $PWD/android-sdk_r24-linux.tgz ] && wget http://dl.google.com/android/android-sdk_r24-linux.tgz && tar xf android-sdk_r24-linux.tgz
```

**Comment by joprice** [comment]
Bash in yaml files is a nightmare. I'm trying moving it into a bash file.

**Comment by stanch** [comment]
Would it work if you place the env exports in the yaml file, instead of moving the `sbt` command into `build.sh`?

**Comment by joprice** [comment]
Yes, but then I'd have to duplicate them. They're needed for android and sbt commands. I guess they have no version in their paths, so they won't change often.

**Comment by stanch** [comment]
> Yes, but then I'd have to duplicate them.

I thought they would be inherited by `bash`...

These caching improvements got out of hand really fast! If you don’t want to waste your time on this, I’m willing to merge the previous changes. Although I have to say I really like the idea of having a separate `prepare-build.sh` script, that’s really useful for setting up the development environment (even if one needs to configure some paths before calling it).

**Comment by joprice** [comment]
Yea, but I think I finally got it working. It really helps a lot to be able to ssh onto the box and debug the builds. I switched to using CircleCI for that feature, plus better vms, faster start times, more stable overall.

**Comment by joprice** [comment]
Should I do one more force push to test the caching?

**Comment by stanch** [comment]
Sure, go ahead. I will merge afterwards. Thanks a lot for your effort!

**Comment by joprice** [comment]
Done. No problem. Happy to help.

**Pull request merged**